nc/views.py

```diff
@@ -169,11 +179,58 @@ def contraband_hit_rate(self, request, pk=None):
         q = Q(search_type__isnull=False)
         self.query(results, group_by=("year", "driver_race", "driver_ethnicity"), filter_=q)
         response["searches"] = results.flatten()
-        # searches
+
+        # contraband
         results = GroupedData(by="year", defaults=GROUP_DEFAULTS)
         q = Q(contraband_found=True)
         self.query(results, group_by=("year", "driver_race", "driver_ethnicity"), filter_=q)
         response["contraband"] = results.flatten()
+
+        # # contraband types
+        qs = Contraband.objects.filter(stop__agency=self.get_object())
+        # # filter down by officer if supplied
+        officer = self.request.query_params.get("officer", None)
+        if officer:
+            qs = qs.filter(stop__officer_id=officer)
+        qs = qs.annotate(
+            year=ExtractYear("stop__date__year"),
+            driver_race=F("person__race"),
+            driver_ethnicity=F("person__ethnicity"),
+            contraband_type=Case(
+                When(Q(ounces__gt=0) | Q(pounds__gt=0) | Q(dosages__gt=0) | Q(grams__gt=0) | Q(kilos__gt=0), then=Value("Drugs")),
+                When(Q(pints__gt=0) | Q(gallons__gt=0), then=Value("Alcohol")),
+                When(Q(money__gt=0), then=Value("Money")),
+                When(Q(weapons__gt=0), then=Value("Weapons")),
+                When(Q(dollar_amount__gt=0), then=Value("Other")),
+                default=Value(""),
+                output_field=CharField()
+            )
+        ).filter(
+            ~Q(contraband_type__exact="")
```

Comment:
I leave out the "Empty" contraband types which didn't have any marked type. I wonder if prod every contraband has a type noted in the objects.